tests/test_petsc.py

```diff
@@ -784,6 +781,30 @@ def test_time_loop():
     assert body4.count('ctx0.t0 = t0') == 1
 
 
+@skipif('petsc')
+def test_solve_output():
+    """
+    Verify that PETScSolve returns the correct output for
+    simple cases e.g with the identity matrix.
+    """
+    grid = Grid(shape=(11, 11), dtype=np.float64)
+
+    u = Function(name='u', grid=grid, space_order=2, dtype=np.float64)
+    v = Function(name='v', grid=grid, space_order=2, dtype=np.float64)
+
+    # Solving Ax=b where A is the identity matrix
+    v.data[:] = 5.0
+    eqn = Eq(u, v)
+    petsc = PETScSolve(eqn, target=u)
+
+    with switchconfig(openmp=False):
+        op = Operator(petsc)
+
+    # Check the solve function returns the correct output
+    op.apply()
```

Comment:
Nitpick: I think it's probably better if the `op.apply()` is also within the `switchconfig`